src/components/ActivityTimeline/index.js

```diff
@@ -1,22 +1,101 @@
+ * The ActivityTimeline displays each of any item's upcoming, current, and past activities in chronological order (ascending or descending).
+ * Notice that ActivityTimeline and TimelineMarker components are related and should be implemented together.
  * @category Layout
  */
 export default function ActivityTimeline(props) {
-    const { children, className, style } = props;
+    const {
+        id,
+        children,
+        className,
+        style,
+        variant,
+        multiple,
+        activeSectionNames,
+        onToggleSection,
+    } = props;
+    const registeredTimelineMarkers = useRef([]);
+    const [activeNames, setActiveNames] = useState(activeSectionNames);
+    const containerRef = useRef();
+
+    useEffect(() => {
+        if (
+            activeSectionNames &&
+            activeSectionNames !== activeNames &&
+            typeof onToggleSection === 'function'
+        ) {
+            setActiveNames(activeSectionNames);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [activeSectionNames, onToggleSection]);
+
+    const privateRegisterMarker = useCallback((stepRef, stepProps) => {
+        if (isChildRegistered(stepProps.name, registeredTimelineMarkers.current)) return;
+        const [...nodes] = getChildTimelineMarkersNodes(containerRef.current);
+        const newStepsList = insertChildOrderly(
+            registeredTimelineMarkers.current,
+            {
+                ref: stepRef,
+                ...stepProps,
+            },
+            nodes,
+        );
+        registeredTimelineMarkers.current = newStepsList;
+    }, []);
+
+    const privateUnregisterMarker = useCallback((stepRef, stepName) => {
+        if (!isChildRegistered(stepName, registeredTimelineMarkers.current)) return;
+        registeredTimelineMarkers.current = registeredTimelineMarkers.current.filter(
+            step => step.name !== stepName,
+        );
+    }, []);
+
+    const privateOnToggleMarker = useCallback(
+        (event, name) => {
+            if (typeof onToggleSection === 'function') {
+                return onToggleSection(event, name);
+            }
+            return setActiveNames(name);
+        },
+        [onToggleSection],
+    );
+
+    const context = useMemo(() => {
+        return {
+            activeNames,
+            multiple,
+            isVariantAccordion: variant === 'accordion',
+            privateRegisterMarker,
+            privateUnregisterMarker,
+            privateOnToggleMarker,
+        };
+    }, [
+        variant,
+        activeNames,
+        multiple,
+        privateRegisterMarker,
+        privateUnregisterMarker,
+        privateOnToggleMarker,
+    ]);
 
     return (
-        <StyledUl className={className} style={style}>
-            {children}
+        <StyledUl id={id} className={className} style={style} ref={containerRef} variant={variant}>
+            <Provider value={context}>{children}</Provider>
         </StyledUl>
```

Comment:
I think here, we should execute all this new code only if the variant is accordion:
```
if (variant === 'accordion') {
   return <NewAccordionComponent />;
}
return <OriginalComponent />;
```